### ISSUE TITLE
[a11y]: Datagrid sort button invalid ARIA attribute

### ISSUE THREAD

**Issue by Jonathan-Maciel** [carbon-design-system/ibm-products#7067]
### Package

@carbon/ibm-products

### Browser

Chrome

### Package version

v2.54.0

### React version

v16.10.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The sort button has the `aria-sort` attribute, but I believe this should be on the parent `<th>`.

![Image](https://github.com/user-attachments/assets/c6f8f095-98a1-4967-bb5d-16f37f1b6679)
![Image](https://github.com/user-attachments/assets/a0378509-8ab0-4332-8226-53f24a001834)

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/deprecated-datagrid-datagrid--sortable-columns&globals=viewport:basic

### Steps to reproduce

Inspecting the linked storybook example, you can see that the `aria-sort` label is applied to the button and not the column head. 

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by wkeese** [comment]
Note that it works better in vanilla Carbon Datagrid, or at least aria-sort is not on the button, see https://react.carbondesignsystem.com/?path=/story/components-datatable-sorting--default.

```html
<th aria-sort="ascending" class="cds--table-sort__header" scope="col">
  <div class="cds--table-sort__description" id="table-sort-:r39:">
    Click to sort rows by Name header in descending order
  </div>
  <button
    type="button"
    aria-describedby="table-sort-:r39:"
    class="cds--table-sort cds--table-sort--active"
  >
    <span class="cds--table-sort__flex"
      ><div class="cds--table-header-label">Name</div>
      <svg ... > ... </svg>
      <svg ... > ... </svg>
      <div class="cds--table-header-label--decorator-inner"></div
    ></span>
  </button>
</th>

{{{

**Comment by Jonathan-Maciel** [comment]
Yes, that `DataTable` component works well most of the time, but for our case we need to have [nested rows similar to this story](https://carbon-for-ibm-products.netlify.app/?path=/story/deprecated-datagrid-datagrid-nestedrows--nested-rows-usage-story&globals=viewport:basic). With the vanilla component there are styling issues with nesting, perhaps related to [the issue you created here](https://github.com/carbon-design-system/carbon/issues/7926).

**Comment by wkeese** [comment]
Hmm, nested tables are different than nested rows, but I don't see why either one would affect the markup or styling used for column headers.

**Comment by elycheea** [comment]
@Jonathan-Maciel @wkeese As you may know, the Datagrid is currently deprecated in favor of [`TanStack/Table`](https://github.com/carbon-design-system/tanstack-carbon), which means support for bugs and enhancements is limited. If you’d like to [contribute a fix](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CONTRIBUTING.md), I’m happy to reopen this issue for you.

**Comment by wkeese** [comment]
Maybe @Jonathan-Maciel will want to do that.  But I looked at useSortableColumns.tsx and it wasn't obvious how to set `aria-sort` on the `TH` instead of the `<Button>`.  It doesn't seem like useSortableColumns.tsx has access to the `TH`.
